pymc/distributions/bound.py

```diff
@@ -182,6 +183,14 @@ def __new__(
         **kwargs,
     ):
 
+        warnings.warn(
+            "Bound has been deprecated in favor of Truncated, and will be removed in a "
+            "future release. If Truncated is not an option, Bound can be implemented by"
+            "adding an IntervalTransform between lower and upper to a continuous "
+            "variable. A Potential that returns negative infinity for values outside "
+            "of the bounds can be used for discrete variables.",
```

Comment:
I dont think using interval transformation is equivalent as a transformation distort the density function. Suggest change the wording to:

```suggestion
            "future release. If you want to exactly replicate the previous behavior of Bound you can"
            "use a Potential that returns negative infinity for values outside "
            "of the bounds.",
```